### DISCUSSION TITLE
"SQLModelConfig" has no attribute "table"

### DISCUSSION THREAD

**Discussion by redb0** [fastapi/sqlmodel#1000] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field, SQLModel


class Foo(SQLModel):
    field_a: str = Field(
        nullable=False,
        description='field A',
    )

print(type(Foo.model_config))  # <-- <class 'dict'>

try:
    print(f'{Foo.model_config.table = }')
except AttributeError:
    print('attr table not found')

print(f'{Foo.model_config.get("table") = }')
```

### Description

VS Code highlights the `table` attribute as `SQLModelConfig`. But in normal use, python outputs that its type is a dictionary. 

![image](https://github.com/tiangolo/sqlmodel/assets/30475117/2b5f9b1d-a2b8-453c-b8a9-78199d5ccbcf)

Is there any way around this other than using `cast`?

```python
cast(dict[str, Any], Foo.model_config)
```

### Operating System

Linux

### SQLModel Version

0.0.19

### Python Version

Python 3.9.19

**Comment by YuriiMotov** [comment, marked as answer]
SQLModel's `model_config` is a `TypedDict` inherited from Pydantic's `BaseConfig` and adds optional `table` and `registry` keys.

Yes, unfortunately IDE fails to correctly handle this..
To safely access this attribute you should use `Foo.model_config.get("table")`
